crates/nu-protocol/src/value/mod.rs

```diff
@@ -2799,100 +2830,99 @@ impl Value {
             }
         }
 
-        fn checked_mod_f64(dividend: f64, divisor: f64) -> Option<f64> {
+        fn checked_div_floor_f64(dividend: f64, divisor: f64) -> Option<f64> {
             if divisor == 0.0 {
                 None
             } else {
-                let remainder = dividend % divisor;
-                if (remainder > 0.0 && divisor < 0.0) || (remainder < 0.0 && divisor > 0.0) {
-                    Some(remainder + divisor)
-                } else {
-                    Some(remainder)
-                }
+                Some((dividend / divisor).floor())
             }
         }
 
         match (self, rhs) {
             (Value::Int { val: lhs, .. }, Value::Int { val: rhs, .. }) => {
-                if let Some(val) = checked_mod_i64(*lhs, *rhs) {
+                if let Some(val) = checked_div_floor_i64(*lhs, *rhs) {
                     Ok(Value::int(val, span))
                 } else if *rhs == 0 {
                     Err(ShellError::DivisionByZero { span: op })
                 } else {
                     Err(ShellError::OperatorOverflow {
-                        msg: "modulo operation overflowed".into(),
+                        msg: "division operation overflowed".into(),
                         span,
                         help: None,
                     })
                 }
             }
             (Value::Int { val: lhs, .. }, Value::Float { val: rhs, .. }) => {
-                if let Some(val) = checked_mod_f64(*lhs as f64, *rhs) {
+                if let Some(val) = checked_div_floor_f64(*lhs as f64, *rhs) {
                     Ok(Value::float(val, span))
                 } else {
                     Err(ShellError::DivisionByZero { span: op })
                 }
             }
             (Value::Float { val: lhs, .. }, Value::Int { val: rhs, .. }) => {
-                if let Some(val) = checked_mod_f64(*lhs, *rhs as f64) {
+                if let Some(val) = checked_div_floor_f64(*lhs, *rhs as f64) {
                     Ok(Value::float(val, span))
                 } else {
                     Err(ShellError::DivisionByZero { span: op })
                 }
             }
             (Value::Float { val: lhs, .. }, Value::Float { val: rhs, .. }) => {
-                if let Some(val) = checked_mod_f64(*lhs, *rhs) {
+                if let Some(val) = checked_div_floor_f64(*lhs, *rhs) {
                     Ok(Value::float(val, span))
                 } else {
                     Err(ShellError::DivisionByZero { span: op })
                 }
             }
             (Value::Filesize { val: lhs, .. }, Value::Filesize { val: rhs, .. }) => {
-                if let Some(val) = checked_mod_i64(*lhs, *rhs) {
-                    Ok(Value::filesize(val, span))
-                } else if *rhs == 0 {
+                if let Some(val) = checked_div_floor_i64(lhs.get(), rhs.get()) {
+                    Ok(Value::int(val, span))
+                } else if *rhs == Filesize::ZERO {
```

Comment:
Since you do quite a bit of this, why not just add a `.is_zero()`? I think it's nicer to read

Reply:
Maybe, seems a little unnecessary though. Besides the filesize-specific methods, I only added methods also present on `i64`.